### PR TITLE
Feat/30 the player styles (UI only)

### PR DESCRIPTION
**Pull request by Cameron933**
### Descriptions
- Link to Zenhub card: https://app.zenhub.com/workspaces/taodemy-63c3dec267d0df73ca6087cb/issues/gh/taodemy/taotify/30
- Link to UI design (if possible): https://www.figma.com/file/aPBUNzoc2uDtxDeLtlQCoz/Taotify?node-id=335-132&t=l9iTUk6G8wG1XkSu-0

This PR already been updated and apply the loop and shuffle button function

### Provide screenshots or videos that the function is working.
Desktop:
image
image

Ipad:
image

Phone:
image

### Provide screenshots of no warning, or error in the console.
image
image

- [x] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket.
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Remove unused comments.

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -127,6 +127,9 @@ module.exports = {
           100: "#A3A3A3",
         },
       },
+      dropShadow: {
+        xsm: "0px 4px 4px rgba(0, 0, 0, 0.25)",
```
Comment (tatoMa) [inline tailwind.config.js:131]: What is xsm means? Xs or sm?

File: src/layouts/index.tsx [hunk 2]
```diff
@@ -11,10 +11,13 @@ const Layout = ({ children }: LayoutProps) => {
   return (
     <div className="flex h-screen w-screen">
       <SideBar />
-      <div className="flex h-full w-[calc(100vw-320px)] flex-col">
+      <div className="flex h-full w-full flex-col md:w-[calc(100vw-64px)] lg:w-[calc(100vw-320px)]">
```
Comment (tatoMa) [inline src/layouts/index.tsx:14]: Nice work.

File: src/layouts/SideBar.tsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import React from "react";
 
 export default function SideBar() {
   return (
-    <aside className="flex h-[100vh] w-[320px] flex-col bg-black text-white">
+    <aside className="hidden h-[100vh] flex-col bg-black text-white md:flex md:w-[64px] lg:flex lg:w-[320px]">
```
Comment (tatoMa) [inline src/layouts/SideBar.tsx:5]: I dont think you need lg:flex cos it has flex since md.

Comment (tatoMa) [inline src/layouts/SideBar.tsx:5]: How about making it looks better? Try to add transition and duration-300 to see how it animates when the width changes.

File: src/layouts/BottomNavBar.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+const BottomNavBar = () => {
+  return (
+    <section className="fixed bottom-[0px] h-[72px] w-full border md:hidden">
+      <div className="relative h-full drop-shadow-xsm">
+        <Image
+          src="/bg_player.png"
+          alt="Player background image"
+          fill
+          style={{ objectFit: "cover" }}
```
Comment (tatoMa) [inline src/layouts/BottomNavBar.tsx:12]: Why use an inline style here?

File: src/components/ui/CoverImage.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    />
+  ) : (
+    <div className="h-16 w-16 md:h-[60px] md:w-[60px] lg:h-[100px] lg:w-[100px]">
+      <Image
+        className="rounded-full"
+        src={src}
+        alt={`Picture of ${variant}`}
+        width={100}
+        height={100}
+        style={{ objectFit: "cover" }}
```
Comment (tatoMa) [inline src/components/ui/CoverImage.tsx:25]: Why use an inline style? Width and height 100% or 100(px)?

File: tailwind.config.js (hunk 1 above)
Comment (Cameron933) [reply tailwind.config.js:131]: Rename it as "bgImgShadow"

File: src/layouts/SideBar.tsx (hunk 3 above)
Comment (Cameron933) [reply src/layouts/SideBar.tsx:5]: Yeah, remove lg:flex

**Review by Trent1900 — APPROVED** [review]
LGTM
the number of file changes might be a bit more than what I expect, but overall is nice work. I will approve that.

File: src/layouts/MusicPlayer.tsx [hunk 6]
```diff
@@ -12,8 +16,38 @@ export default function MusicPlayer() {
...
   return (
-    <div className="fixed bottom-0 h-[120px] w-[calc(100vw-320px)] border px-4">
-      <audio src={audioUrl} role="audio" controls autoPlay onEnded={handleEnd}></audio>
-    </div>
+    <section className="fixed bottom-[72px] h-[78px] w-full md:bottom-0 md:h-[120px] md:w-[calc(100vw-64px)] lg:w-[calc(100vw-320px)]">
+      <div className="relative h-full drop-shadow-bgImgShadow">
+        <Image src="/bg_player.png" alt="Player background image" fill />
+      </div>
+
+      <div className="absolute left-0 top-0 flex h-full w-full gap-2 bg-dark-400 bg-opacity-80 px-2 backdrop-blur-2xl md:px-4 md:py-2">
```
Comment (Double-Twelve) [inline src/layouts/MusicPlayer.tsx:24]: should add `md:gap-4`

File: src/layouts/SideBar.tsx (hunk 3 above)
Comment (Cameron933) [reply src/layouts/SideBar.tsx]: Problem solved by using: "transition-all"

File: src/layouts/MusicPlayer.tsx [hunk 7]
```diff
@@ -12,8 +16,38 @@ export default function MusicPlayer() {
...
+            <p className="text-sm">By Jaxson Westervelt</p>
+          </div>
+        </div>
+
+        <div className="hidden items-start justify-center gap-[10px] px-2 text-light lg:flex lg:flex-col ">
+          <p className="text-base">This is love</p>
+          <p className="text-sm">By Jaxson Westervelt</p>
+        </div>
+
+        <div className="flex flex-grow flex-col items-center gap-1 self-stretch py-1 md:items-start md:gap-2 md:p-[10px]">
```
[the remark this answers is not in the document]
Comment (Cameron933) [reply src/layouts/MusicPlayer.tsx:39]: 1. deleted "self stretch"
2. added "md:justify-center". Besides, I just checked the UI and md:gap-2 is correct
3. removed "md:item-start". Besides, md:p-[10px] is correct
image

File: src/components/AudioControls.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+const AudioControls = ({
+  isPlaying,
+  onPlayPauseClick,
+  onPrevClick,
+  onNextClick,
+}: AudioControlsProps) => {
+  return (
+    <div className="flex h-16 w-full items-center justify-between text-light">
```
[the remark this answers is not in the document]
Comment (Cameron933) [reply src/components/AudioControls.tsx:22]: Just checked the UI, the layout will not be same without "h-16"

File: src/components/AudioControls.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+      <button type="button" className="flex h-8 w-8 items-center justify-center">
```
[the remark this answers is not in the document]
Comment (Cameron933) [reply src/components/AudioControls.tsx:23]: I just tried it, I think keep it as a background will be better for the accessibility, which makes the user easier to click it

File: src/components/AudioControls.tsx [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        <BiHeart className="h-6 w-6" />
+      </button>
+
+      <div className="flex w-full items-center gap-2">
```
[the remark this answers is not in the document]
Comment (Cameron933) [reply src/components/AudioControls.tsx:27]: fixed

**Review by yunshen513 — APPROVED** [review]
LGTM

**Pull request merged**